octobot_tentacles_manager/base_worker/tentacle_worker.py

```diff
@@ -0,0 +1,199 @@
+        # tentacles folder
+        found_existing_installation = not await find_or_create(self.tentacle_path)
+        # tentacle mail python init file
+        await find_or_create(path.join(self.tentacle_path, PYTHON_INIT_FILE), False,
+                             self._get_module_init_file_content(TENTACLES_FOLDERS_ARCH.keys()))
+        # tentacle inner architecture
+        await self._rec_create_missing_files(self.tentacle_path, TENTACLES_FOLDERS_ARCH)
+        return found_existing_installation
+
+    @staticmethod
+    async def _rec_create_missing_files(root_folder, files_arch):
+        sub_dir_to_create_coroutines = []
+        for root, subdir in files_arch.items():
+            current_root = path.join(root_folder, root)
+            await find_or_create(current_root)
+            if isinstance(subdir, dict):
+                # create python init file
+                await find_or_create(path.join(current_root, PYTHON_INIT_FILE), False,
+                                     TentacleWorker._get_module_init_file_content(subdir))
+                sub_dir_to_create_coroutines.append(TentacleWorker._rec_create_missing_files(current_root, subdir))
+            else:
+                # create python init file
+                await find_or_create(path.join(current_root, PYTHON_INIT_FILE), False,
+                                     TentacleWorker._get_module_init_file_content(subdir))
+                sub_dir_to_create_coroutines += [find_or_create(path.join(current_root, directory))
+                                                 for directory in subdir]
+        await gather(*sub_dir_to_create_coroutines)
+
+    @staticmethod
+    def _get_module_init_file_content(modules):
+        return "\n".join(f"from .{module} import *" for module in modules)
+
+    def _find_tentacles_missing_requirements(self, tentacle_data, version_by_modules):
+        missing_requirements = {}
+        if tentacle_data.tentacles_requirements:
+            # check if requirement is in tentacles to be installed in this call
+            for requirement, version in tentacle_data.extract_tentacle_requirements():
+                if not self._is_requirement_satisfied(requirement, version, tentacle_data, version_by_modules):
+                    missing_requirements[requirement] = version
+        return missing_requirements
+
+    def _is_requirement_satisfied(self, requirement, version, tentacle_data, version_by_modules):
+        satisfied = False
+        if requirement in version_by_modules:
+            available = version_by_modules[requirement]
+            if version is None:
+                satisfied = True
+            elif version != available:
+                self.logger.error(f"Incompatible tentacle version requirement for "
+                                  f"{tentacle_data.name}: requires {version}, installed: "
+                                  f"{available}. This tentacle might not work as expected")
+                satisfied = True
```

Comment:
`satisfied = True` ?

Reply:
For now we can't download a tentacle requirement with a specific version, therefore we consider this case as satisfied to avoid re-downloading the same just downloaded tentacle folder in case of version conflicts